newsroom/users/users.py

```diff
@@ -19,6 +20,32 @@
 from newsroom.companies.utils import get_company_section_names, get_company_product_ids
 
 
+class UserAuthentication(SessionAuth):
+    def authorized(self, allowed_roles, resource, method):
+        if super().authorized(allowed_roles, resource, method):
+            return True
+
+        if not request.view_args or not request.view_args.get("_id"):
+            # not a request for a specific user, stop
+            return False
+
+        if request.view_args["_id"] == str(get_user_id()):
+            # current user editing current user
+            return True
+
+        current_user = get_user()
+        if not current_user.get("company") or current_user.get("user_type") != UserRole.COMPANY_ADMIN.value:
```

Comment:
Should we also allow system admins to modify users?

Reply:
that's handled by the initial `if authorized(): return True`